Sources/SwiftHttp/HttpUrl.swift

```diff
@@ -137,10 +150,12 @@ public extension HttpUrl {
         var path = "/" + path.joined(separator: "/")
         if let resource = resource {
             path += (resource.hasPrefix("/") ? resource : "/" + resource)
-        }
-        else {
+        } else {
             path += "/"
         }
+				if path.last == "/", !query.isEmpty {
```

Comment:
Why do you trim the trailing slash? 🤔 The HttpUrlTests are failing because of this change...

Reply:
I came across the fact that some tools define the url as invalid if it contains "path/?...", so the valid url end with "path?..."

Reply:
please take a look at `isTrailingSlashEnabled` in HttpUrl